### PR TITLE
Seventh day

### PR DESCRIPTION
**Pull request by hrodberaht**

### CODE REVIEW

File: src/components/AllCards/AllCards.js [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
+import PropTypes from 'prop-types';
+import Card from './Card/Card';
 
 
 export default class AllCards extends Component {
-  
   render() {
-    const {  cards, handleClick, filterText } = this.props;
+    const { cards, handleClick, filterText } = this.props;
     let list = cards;
```
Comment (ar-mac) [inline src/components/AllCards/AllCards.js:9]: Personally I find using `let` a last resort, as you need to pay special attention to what actually is in the variable. In this case we can extract this to the method.

```
cards = () => {
  const { cards, filterText } = this.props;
  if (filterText) {
    const regexp = new RegExp(filterText, 'i')
    return cards.filter(card => regexp.test(card.name);
  }
  return cards;
}
```

and then use it in `return`
`{cards().map(card => {`

File: src/components/AllCards/AllCards.js [hunk 2]
```diff
@@ -1,31 +1,25 @@
...
     if (filterText) {
-      list = cards.filter((card) => {
-          return card.name.toLowerCase().includes(filterText);
-      });
+      list = cards.filter(card => card.name.toLowerCase().includes(filterText));
     }
 
     return (
       <div>
```
Comment (ar-mac) [inline src/components/AllCards/AllCards.js:15]: If you do not need this div for styling/positional purposes you can ommit it using `<React.Fragment>` or just `<>` tag. https://reactjs.org/docs/fragments.html

File: src/components/ShowAddCard/ShowAddCard.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { configure, shallow, mount } from "enzyme";
+import React from "react";
+import Adapter from "enzyme-adapter-react-16";
+import ShowAddCard from "./ShowAddCard";
+import AddCardForm from "../../containers/AddCardForm";
+
+configure({ adapter: new Adapter() });
```
Comment (ar-mac) [inline src/components/ShowAddCard/ShowAddCard.test.js:7]: Adapter configuration should be on separate, automatically loaded file. Unless you want to configure adapter in the every file.

I cannot find any description how to do it, so you can base on the jest setup done in my app done during coderDojo
https://github.com/pokiujf/emoji_roulette/pull/9/files
Especially file `src/setupTests.js` It's automatically loaded by create-rect-app so you just define initialization files to call.

Also worth noting is `src/specs/buildSetup.js` It's a pattern developped by us in Selleo, to simplify testing setup. Initially do not use this pattern, and render components in specs as it comes to you naturally. So later when you switch to buildSetup you'll see the difference.

File: src/reducers/cards.js [hunk 4]
```diff
@@ -16,8 +16,7 @@ const cards = (state = initialState, actions) => {
     case "ADD_CARD":
       return [...state,{id: actions.id, name: actions.name,email: actions.email}];
     case "DELETE_CARD":
-      const index = state.findIndex(card => card.id === actions.id);
-      return [...state.slice(0, index), ...state.slice(index + 1)]
+      return state.filter((card) => card.id !== actions.id)
```
Comment (ar-mac) [inline src/reducers/cards.js:19]: 👍

File: src/components/AllCards/AllCards.js (hunk 1 above)
Comment (hrodberaht) [reply src/components/AllCards/AllCards.js:9]: ok, i understand what you mean

File: src/components/ShowAddCard/ShowAddCard.test.js (hunk 3 above)
Comment (hrodberaht) [reply src/components/ShowAddCard/ShowAddCard.test.js:7]: thx for you solution

File: src/components/AllCards/AllCards.js (hunk 2 above)
Comment (hrodberaht) [reply src/components/AllCards/AllCards.js:15]: i have a problem with <>, i think is problem with babel, i will look for solution for that

Comment (ar-mac) [reply src/components/AllCards/AllCards.js:15]: Go ahead 👍

**Pull request merged**